app/src/main/java/cm/aptoide/pt/home/HomePresenter.java

```diff
@@ -185,25 +185,39 @@ private void handlePromotionsClick() {
         .filter(lifecycleEvent -> lifecycleEvent.equals(View.LifecycleEvent.CREATE))
         .observeOn(viewScheduler)
         .doOnNext(created -> view.showLoading())
-        .flatMapSingle(created -> loadBundles())
+        .flatMapSingle(__ -> loadHome())
         .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
         .subscribe(__ -> {
         }, crashReporter::log);
   }
 
-  @NonNull private Single<HomeBundlesModel> loadBundles() {
-    return home.loadHomeBundles()
+  private Single<Boolean> showNativeAds() {
+    return home.shouldLoadNativeAd()
+        .observeOn(viewScheduler)
+        .doOnSuccess(showNatives -> view.setAdsTest(showNatives));
+  }
+
+  private Single<HomeBundlesModel> loadHome() {
+    return Single.zip(showNativeAds(), loadBundles(), (aBoolean, bundlesModel) -> bundlesModel)
         .observeOn(viewScheduler)
         .doOnSuccess(bundlesModel -> {
-          if (bundlesModel.hasErrors()) {
-            handleError(bundlesModel.getError());
-          } else if (!bundlesModel.isLoading()) {
-            view.hideLoading();
-            view.showBundles(bundlesModel.getList());
-          }
+          handleLoadBundlesResult(bundlesModel);
         });
   }
 
+  @NonNull private Single<HomeBundlesModel> loadBundles() {
+    return home.loadHomeBundles();
+  }
+
+  private void handleLoadBundlesResult(HomeBundlesModel bundlesModel) {
```

Comment:
name a bit weird, handleBundlesResult is my suggestion - but I don't think it is much better either.